funfact/test_loss.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import pytest  # noqa: F401
+from funfact import active_backend as ab
+from .loss import (
+    MSE,
+    L1,
+    KLDivergence,
+)
+
+
+@pytest.mark.parametrize('loss_cls', [
+    MSE,
+    L1,
+])
+def test_seminorms(loss_cls):
+
+    loss = loss_cls()
+    for shape in [(3,), (3, 3), (3, 4), (3, 4, 5), (2, 3, 4, 5)]:
+        for _ in range(64):
+            a = ab.normal(0.0, 1.0, shape)
+            b = ab.normal(0.0, 1.0, shape)
+            assert loss(a, a) == pytest.approx(0, abs=1e-6)
+            assert loss(b, b) == pytest.approx(0, abs=1e-6)
+            assert loss(a, b) >= 0
+            assert loss(b, a) >= 0
```

Comment:
Maybe we can add a test that `loss(a, b)` and `loss(b,a)` should be approximately equal?

Reply:
Good point.

Reply:
Yes, but I believe this test only runs for Lp norms :)

Reply:
Done

Reply:
> Yes, but I believe this test only runs for Lp norms :)

Yeah, I realized it as soon as I opened up the file. Seems like I already forgot part of what I wrote two days ago. :joy: